Source/Tests/LinuxSyscalls/x64/Memory.cpp

```diff
@@ -47,6 +51,9 @@ namespace FEX::HLE::x64 {
 
     REGISTER_SYSCALL_IMPL_X64(mprotect, [](FEXCore::Core::InternalThreadState *Thread, void *addr, size_t len, int prot) -> uint64_t {
       uint64_t Result = ::mprotect(addr, len, prot);
+      if (Result != -1 && prot & PROT_EXEC) {
```

Comment:
Thinking about this. Shouldn't we also be flushing when the region was previously PROT_EXEC and the new mprotect is not PROT_EXEC?

Otherwise a program running under fex could theoretically continue executing cached code on a non-executable page. Maybe there is some form of jit out there that relies on segfaults from stale code to make sure they are removed from block-linking? 

Then again, I'm pretty sure our current jit doesn't even check page executablity before emitting code. Should we create an issue and push this down the line?